util/test.py

```diff
@@ -267,6 +258,7 @@ def validate_output(self, out):
 
   def fail(self, message, *args):
     if args:
+      args = [arg.decode("utf-8") if type(arg) is bytes else arg for arg in args]
```

Comment:
This is the bit I'm least happy about. It's not critical, but without it errors are printed as python bytes literals, e.g. `b'foo'`.